assets/javascripts/discourse/components/channels-list.js

```diff
@@ -37,6 +37,21 @@ export default class ChannelsList extends Component {
     }`;
   }
 
+  @computed(
+    "publicChannelsEmpty",
+    "currentUser.{staff,has_joinable_public_channels}"
+  )
+  get displayPublicChannels() {
+    if (this.publicChannelsEmpty) {
+      return (
+        this.currentUser?.staff ||
+        this.currentUser?.has_joinable_public_channels
```

Comment:
I wonder if we should make the server return true for "has joinable public channels" for staff members so we don't have to check both on the client?

Reply:
Yes I hesitated on this, my current thinking is that we want to keep `has_joinable_public_channels` "correct". You might be a staff and have followed all the existing channels and we want to return false in this case otherwise this would be quite unexpected.